src/main/java/com/hierynomus/spnego/NegTokenInit.java

```diff
@@ -102,11 +103,18 @@ public NegTokenInit read(byte[] bytes) throws IOException {
     public NegTokenInit read(Buffer<?> buffer) throws IOException {
         try {
             ASN1Primitive applicationSpecific = new ASN1InputStream(buffer.asInputStream()).readObject();
-            if (!(applicationSpecific instanceof ASN1ApplicationSpecific)) {
+            if (!(applicationSpecific instanceof BERApplicationSpecific || applicationSpecific instanceof DERApplicationSpecific)) {
```

Comment:
ok, agreed. I actually want to see whether for the ASN.1 part we can't use smetihing else, as it is the only hardcoded dependency on BC. But that is for a later date :)